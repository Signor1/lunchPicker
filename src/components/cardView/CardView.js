import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useHistory } from "react-router-dom";
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import StarIcon from '@material-ui/icons/Star';
import FavoriteIcon from '@material-ui/icons/Favorite';
import LinkIcon from '@material-ui/icons/Link';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { red } from '@material-ui/core/colors';
import { yellow } from '@material-ui/core/colors';
import _ from 'lodash';
import axios from 'axios';

import { getRootUrl, log } from '../../common/Common';

const ROOT_URL = getRootUrl();

const useStyles = makeStyles(theme => ({
  card: {
    maxWidth: 345,
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

function CardView(props) {
  const classes = useStyles();
  const history = useHistory();

  const [expanded, setExpanded] = useState(false);
  const [reviewsList, setReviewsList] = useState([]);

  const item = props.resultListItem;
  const id = item.id;
  const name = item.name;
  const avatarStr = name[0].toUpperCase();
  const categories = item.categories;
  const imageUrl = item.image_url;
  const url = item.url;
  const rating = item.rating;
  const location = item.location.display_address.join(', ');
  const displayPhone = item.display_phone;

  let subHeader = "";
  if (!_.isEmpty(categories)) {
    categories.forEach((item, i) => {
      if (i === 0)
        subHeader += item.title;
      else
        subHeader += ", " + item.title;
    });
  }

  const handleExpandClick = () => {
    setExpanded(!expanded);

    if (_.isEmpty(reviewsList))
      getRestaurantsDetailsReviewById(id);
  };

  const getRestaurantsDetailsReviewById = () => {
    axios.get(
      `${ROOT_URL}/restaurant/get-restaurant-details-review/${id}`,
      {
        headers: {
          'Content-Type': 'application/json'
        }
      }
    )
      .then((response) => {
        if (!_.isEmpty(response)) {
          log("response = ", response);
          const reviewsList = response.data.restaurantDetailsReview.reviews;
          setReviewsList(reviewsList);
        }
      })
      .catch((error) => {
        if (!_.isEmpty(error)) {
          log("error = ", error);
        }
      });
  };

  const handleOpenUrl = () => {
    window.open(url);
  }

  const renderStarIcon = () => {
    let starIconList = [];

    for (let i = 0; i < rating; i++) {
      starIconList.push(
        <StarIcon key={i} style={{ color: yellow[600] }} fontSize="small" />
      );
    }

    return starIconList;
  }

  const handleOpenUserProfileUrl = (userProfileUrl) => {
    window.open(userProfileUrl);
  }

  const renderReviewsList = () => {
    let formattedReviewsList = null;

    if (!_.isEmpty(reviewsList)) {
      formattedReviewsList = reviewsList.map((item, i) => {
        const text = item.text;
        const userName = item.user.name;
        const userProfileUrl = item.user.profile_url;
        const userImageUrl = item.user.image_url;
        return (
          <div key={i} className="my-3">
            <Avatar style={{ cursor: 'pointer' }} alt="user" src={userImageUrl} onClick={() => handleOpenUserProfileUrl(userProfileUrl)} />
            <Typography className="mt-2">
              <b>{userName}</b>: {text}
            </Typography>
          </div>
        )
      });
    }

    return (
      <div>
        <Typography paragraph>Review:</Typography>
        {formattedReviewsList}
      </div>
    );
  }

  const handleLocationClick = (e) => {
    const text = e.target.innerHTML;
    window.open(`https://www.google.com/maps/search/?api=1&query=${text}`);
  }

  const handleOnMouseEnterTextStyle = (e) => {
    e.target.setAttribute('style', 'cursor: pointer; text-decoration: underline; color: #ed1f30');
  }

  const handleOnMouseLeaveTextStyle = (e) => {
    e.target.removeAttribute('style');
  }

  const handleOpenRestaurantDetailsById = (id) => {
    history.push(`/restaurant-details/${id}`);
  }

  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={
          <Avatar
            style={{ cursor: 'pointer' }}
            aria-label="recipe"
            className={classes.avatar}
            onClick={() => handleOpenRestaurantDetailsById(id)}>
            {avatarStr}
          </Avatar>
        }
        action={
          <IconButton aria-label="settings">
            <MoreVertIcon />
          </IconButton>
        }
        title={
          <div
            onClick={() => handleOpenRestaurantDetailsById(id)}
            onMouseEnter={(e) => handleOnMouseEnterTextStyle(e)}
            onMouseLeave={(e) => handleOnMouseLeaveTextStyle(e)}>
            {name}
          </div>
        }
        subheader={subHeader}
      />
      <CardMedia
        style={{ cursor: 'pointer' }}
        className={classes.media}
        image={imageUrl}
        title={name}
        onClick={handleOpenUrl}
      />
      <CardContent>
        <Typography
          variant="body2"
          color="textSecondary"
          component="p">
          Location: <span
            onClick={(e) => handleLocationClick(e)}
            onMouseEnter={(e) => handleOnMouseEnterTextStyle(e)}
            onMouseLeave={(e) => handleOnMouseLeaveTextStyle(e)}>
            {location}
          </span>
        </Typography>
        <div className="my-2"></div>
        <Typography variant="body2" color="textSecondary" component="p">
          {!_.isEmpty(displayPhone) ? `Phone: ${displayPhone}` : ''}
        </Typography>
        <div className="my-2"></div>
        {renderStarIcon()}
      </CardContent>
      <CardActions disableSpacing>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon style={{ color: red[500] }} />
        </IconButton>
        <IconButton aria-label="link" onClick={handleOpenUrl}>
          <LinkIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          {renderReviewsList()}
        </CardContent>
      </Collapse>
    </Card>
  );
}

export default CardView;
