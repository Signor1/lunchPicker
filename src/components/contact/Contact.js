import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Tooltip from '@material-ui/core/Tooltip';
import EmailIcon from '@material-ui/icons/Email';
import GitHubIcon from '@material-ui/icons/GitHub';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3, 2),
  },
}));

function Contact() {
  const classes = useStyles();

  const handleEmailIconClick = () => {
    window.open('mailto:yeukfei02@gmail.com');
  }

  const handleGithubIconClick = () => {
    window.open('https://github.com/yeukfei02');
  }

  return (
    <div>
      <div className="mt-5 d-flex justify-content-center">
        <Paper className={`${classes.root} mx-4 w-75 text-center`}>
          <h4>contact us via email or visit our github repo</h4>
          <Tooltip title="yeukfei02@gmail.com" placement="bottom">
            <EmailIcon style={{ cursor: 'pointer' }} className="mr-3" fontSize="large" onClick={handleEmailIconClick} />
          </Tooltip>
          <Tooltip title="yeukfei02" placement="bottom">
            <GitHubIcon style={{ cursor: 'pointer' }} fontSize="large" onClick={handleGithubIconClick} />
          </Tooltip>
        </Paper>
      </div>
      <div className="my-5 d-flex justify-content-center">
        <Paper className={`${classes.root} mx-4 w-75 text-center`}>
          <a
            className="dbox-donation-button"
            style={{ background: '#ff0000', color: '#fff', textDecoration: 'none', fontFamily: 'Verdana,sans-serif', display: 'inline-block', fontSize: '16px', padding: '13px 17px', borderRadius: '2px', boxShadow: '0 1px 0 0 #1f5a89', textShadow: '0 1px rgba(0, 0, 0, 0.3)' }}
            href="https://donorbox.org/donate-for-lunch-picker-better-features-and-development?default_interval=o">
            Donate
          </a>
        </Paper>
      </div>
    </div>
  )
}

export default Contact;
