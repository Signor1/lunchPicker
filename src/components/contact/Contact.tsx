import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Tooltip from '@material-ui/core/Tooltip';
import EmailIcon from '@material-ui/icons/Email';
import GitHubIcon from '@material-ui/icons/GitHub';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import Button from '@material-ui/core/Button';
import { StripeProvider } from 'react-stripe-elements';
import MyStoreCheckout from '../myStoreCheckout/MyStoreCheckout';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';

import { getStripeApiKey } from '../../helpers/helpers';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3, 2),
  },
  formControl: {
    margin: theme.spacing(3),
  },
}));

function Contact(): JSX.Element {
  const classes = useStyles();
  const { t } = useTranslation();

  const [radioButtonValue, setRadioButtonValue] = useState<string>('stripe');

  const handleEmailIconClick = () => {
    window.open('mailto:yeukfei02@gmail.com');
  };

  const handleGithubIconClick = () => {
    window.open('https://github.com/yeukfei02');
  };

  const handleDonate = () => {
    window.open('https://donorbox.org/donate-for-lunch-picker-better-features-and-development');
  };

  const handleBuyMeACoffee = () => {
    window.open('https://www.buymeacoffee.com/yeukfei02');
  };

  const handleRadioButtonChange = (e: any) => {
    setRadioButtonValue(e.target.value);
  };

  const renderDiv = () => {
    let resultDiv: any = null;

    if (_.isEqual(radioButtonValue, 'donorbox')) {
      resultDiv = (
        <div>
          <Button className="w-100" variant="outlined" color="primary" onClick={handleDonate}>
            {t('donate')}
          </Button>
        </div>
      );
    } else if (_.isEqual(radioButtonValue, 'buyMeACoffee')) {
      resultDiv = (
        <div>
          <Button className="w-100" variant="outlined" color="primary" onClick={handleBuyMeACoffee}>
            {t('buyMeACoffee')}
          </Button>
        </div>
      );
    } else if (_.isEqual(radioButtonValue, 'stripe')) {
      const stripeApiKey = getStripeApiKey() as string;
      resultDiv = (
        <div>
          <StripeProvider apiKey={stripeApiKey}>
            <MyStoreCheckout />
          </StripeProvider>
        </div>
      );
    }

    return resultDiv;
  };

  return (
    <div>
      <div className="mt-5 d-flex justify-content-center">
        <Paper className={`${classes.root} mx-4 w-75 text-center`}>
          <h5>{t('contactTitle')}</h5>
          <Tooltip title="yeukfei02@gmail.com" placement="bottom">
            <EmailIcon style={{ cursor: 'pointer' }} className="mr-3" fontSize="large" onClick={handleEmailIconClick} />
          </Tooltip>
          <Tooltip title="yeukfei02" placement="bottom">
            <GitHubIcon style={{ cursor: 'pointer' }} fontSize="large" onClick={handleGithubIconClick} />
          </Tooltip>
        </Paper>
      </div>
      <div className="my-5 d-flex justify-content-center">
        <Paper className={`${classes.root} mx-4 w-75`}>
          <h5 className="text-center">{t('donateTitle')}</h5>
          <FormControl component="fieldset" className={classes.formControl}>
            <RadioGroup
              aria-label="position"
              name="position"
              value={radioButtonValue}
              onChange={handleRadioButtonChange}
            >
              <FormControlLabel
                value="donorbox"
                control={<Radio color="primary" />}
                label="Donorbox"
                labelPlacement="end"
              />
              <FormControlLabel
                value="buyMeACoffee"
                control={<Radio color="primary" />}
                label="Buy me a coffee"
                labelPlacement="end"
              />
              <FormControlLabel
                value="stripe"
                control={<Radio color="primary" />}
                label="Stripe"
                labelPlacement="end"
              />
            </RadioGroup>
          </FormControl>
          {renderDiv()}
        </Paper>
      </div>
    </div>
  );
}

export default Contact;
