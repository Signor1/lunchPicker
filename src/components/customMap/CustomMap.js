import React from 'react';
import Map from 'pigeon-maps'
import Marker from 'pigeon-marker';

function CustomMap(props) {
  return (
    <div style={{ overflowX: 'hidden' }}>
      <Map
        center={[props.latitude || 0, props.longitude || 0]}
        zoom={13}
        width={700}
        height={450}>
        <Marker anchor={[props.latitude || 0, props.longitude || 0]} payload={1} onClick={({ event, anchor, payload }) => { }} />
      </Map>
    </div>
  )
}

export default CustomMap;
