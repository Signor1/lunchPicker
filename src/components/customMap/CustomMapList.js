import React, { useState, useEffect } from 'react';
import { Map, TileLayer, Marker, Popup } from 'react-leaflet';
import _ from 'lodash';

function CustomMapList(props) {
  const [markerList, setMarkerList] = useState([]);

  useEffect(() => {
    setMarkerList([]);
    if (!_.isEmpty(props.nameList) && !_.isEmpty(props.locationStrList) && !_.isEmpty(props.coordinatesList)) {
      const marketList = _.merge(props.nameList, props.locationStrList, props.coordinatesList);
      setMarkerList(marketList);
    }
  }, [props.nameList, props.locationStrList, props.coordinatesList]);

  const position = [props.latitude || 0, props.longitude || 0];

  const handleAddressClick = (e) => {
    const text = e.target.innerHTML;
    window.open(`https://www.google.com/maps/search/?api=1&query=${text}`);
  }

  const handleOnMouseEnterTextStyle = (e) => {
    e.target.setAttribute('style', 'cursor: pointer; text-decoration: underline; color: #ed1f30');
  }

  const handleOnMouseLeaveTextStyle = (e) => {
    e.target.removeAttribute('style');
  }

  const renderMarker = () => {
    let renderMarkerList = [];

    if (!_.isEmpty(markerList)) {
      renderMarkerList = markerList.map((item, i) => {
        const position = [item.latitude || 0, item.longitude || 0];
        return (
          <Marker key={i} position={position}>
            <Popup>
              <span><b>{item.name}</b></span>
              <br />
              <span
                onClick={(e) => handleAddressClick(e)}
                onMouseEnter={(e) => handleOnMouseEnterTextStyle(e)}
                onMouseLeave={(e) => handleOnMouseLeaveTextStyle(e)}>
                {item.locationStr}
              </span>
            </Popup>
          </Marker>
        )
      });
    }

    return renderMarkerList;
  }

  return (
    <Map center={position} zoom={13} style={{ width: "100%", height: "600px" }}>
      <TileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
      />
      {renderMarker()}
    </Map>
  )
}

export default CustomMapList;
