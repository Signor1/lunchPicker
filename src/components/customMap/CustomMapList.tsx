import React, { useState, useEffect } from 'react';
import { Map, TileLayer, Marker, Popup } from 'react-leaflet';
import { LatLngTuple } from 'leaflet';
import _ from 'lodash';

function CustomMapList(props: any): JSX.Element {
  const [markerList, setMarkerList] = useState<any[]>([]);

  useEffect(() => {
    setMarkerList([]);
    if (!_.isEmpty(props.nameList) && !_.isEmpty(props.locationStrList) && !_.isEmpty(props.coordinatesList)) {
      const marketList = _.merge(props.nameList, props.locationStrList, props.coordinatesList);
      setMarkerList(marketList);
    }
  }, [props.nameList, props.locationStrList, props.coordinatesList]);

  const position = [props.latitude || 0, props.longitude || 0] as LatLngTuple;

  const handleAddressClick = (e: any) => {
    const text = e.target.innerHTML;
    window.open(`https://www.google.com/maps/search/?api=1&query=${text}`);
  };

  const handleOnMouseEnterTextStyle = (e: any) => {
    e.target.setAttribute('style', 'cursor: pointer; text-decoration: underline; color: #ed1f30');
  };

  const handleOnMouseLeaveTextStyle = (e: any) => {
    e.target.removeAttribute('style');
  };

  const renderMarker = () => {
    let renderMarkerList: any[] = [];

    if (!_.isEmpty(markerList)) {
      renderMarkerList = markerList.map((item: any, i: number) => {
        const position = [item.latitude || 0, item.longitude || 0] as LatLngTuple;
        return (
          <Marker key={i} position={position}>
            <Popup>
              <span>
                <b>{item.name}</b>
              </span>
              <br />
              <span
                onClick={e => handleAddressClick(e)}
                onMouseEnter={e => handleOnMouseEnterTextStyle(e)}
                onMouseLeave={e => handleOnMouseLeaveTextStyle(e)}
              >
                {item.locationStr}
              </span>
            </Popup>
          </Marker>
        );
      });
    }

    return renderMarkerList;
  };

  return (
    <Map center={position} zoom={13} style={{ width: '100%', height: '600px' }}>
      <TileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
      />
      {renderMarker()}
    </Map>
  );
}

export default CustomMapList;
