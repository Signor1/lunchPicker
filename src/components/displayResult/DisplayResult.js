import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import _ from 'lodash';

import CardView from '../cardView/CardView';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3, 2),
    flexGrow: 1,
  },
}));

function DisplayResult(props) {
  const classes = useStyles();

  const renderDisplayResult = () => {
    let cardViewResultList = null;

    if (!_.isEmpty(props.resultList)) {
      cardViewResultList = props.resultList.map((item, i) => {
        return (
          <Grid key={i} item xs={12} sm={4}>
            <div className="d-flex justify-content-center">
              <CardView resultListItem={item} />
            </div>
          </Grid>
        );
      });
    }

    return cardViewResultList;
  }

  const renderDiv = () => {
    let renderDiv = null;

    if (!_.isEmpty(props.resultList)) {
      renderDiv = (
        <div className={classes.root} style={{ overflow: 'hidden' }}>
          <Grid container spacing={3}>
            {renderDisplayResult()}
          </Grid>
        </div>
      );
    }

    return renderDiv;
  }

  return (
    renderDiv()
  )
}

export default DisplayResult;
