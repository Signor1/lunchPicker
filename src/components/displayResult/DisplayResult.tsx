import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Fade from 'react-reveal/Fade';
import _ from 'lodash';

import CardView from '../cardView/CardView';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(3, 2),
    flexGrow: 1,
  },
}));

function DisplayResult(props: any) {
  const classes = useStyles();

  const renderDisplayResult = () => {
    let cardViewResultList = null;

    if (!_.isEmpty(props.resultList)) {
      cardViewResultList = props.resultList.map((item: any, i: number) => {
        return (
          <Grid key={i} item xs={12} sm={4}>
            <Fade bottom>
              <div className="d-flex justify-content-center">
                <CardView resultListItem={item} />
              </div>
            </Fade>
          </Grid>
        );
      });
    }

    return cardViewResultList;
  };

  const renderDiv = () => {
    let renderDiv: any = null;

    if (!_.isEmpty(props.resultList)) {
      renderDiv = (
        <div className={classes.root} style={{ overflow: 'hidden' }}>
          <Grid container spacing={3}>
            {renderDisplayResult()}
          </Grid>
        </div>
      );
    }

    return renderDiv;
  };

  return renderDiv();
}

export default DisplayResult;
