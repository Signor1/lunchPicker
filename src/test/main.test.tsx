import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';

import NavBar from '../components/navBar/NavBar';
// import MainPage from '../components/mainPage/MainPage';
import RandomFood from '../components/randomFood/RandomFood';
import RandomFoodMapView from '../components/randomFoodMapView/RandomFoodMapView';
import Favourites from '../components/favourites/Favourites';
import Settings from '../components/settings/Settings';
import Contact from '../components/contact/Contact';
import RestaurantDetails from '../components/restaurantDetails/RestaurantDetails';

import FloatingActionButton from '../components/floatingActionButton/FloatingActionButton';
import Snackbar from '../components/snackBar/SnackBar';
import DisplayResult from '../components/displayResult/DisplayResult';
import CardView from '../components/cardView/CardView';
import CustomMapList from '../components/customMap/CustomMapList';
import MyStoreCheckout from '../components/myStoreCheckout/MyStoreCheckout';
import ReactTable from '../components/reactTable/ReactTable';
import ImageSlider from '../components/imageSlider/ImageSlider';
import CustomMap from '../components/customMap/CustomMap';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: any) => key,
    i18n: (key: any) => key,
  }),
}));

describe('main.test', () => {
  describe('render test', () => {
    it('NavBar', () => {
      const wrapper = shallow(<NavBar />);
      expect(wrapper).toMatchSnapshot();
    });

    // it('MainPage', () => {
    //   const wrapper = shallow(<MainPage />);
    //   expect(wrapper).toMatchSnapshot();
    // });

    it('RandomFood', () => {
      const wrapper = shallow(<RandomFood />);
      expect(wrapper).toMatchSnapshot();
    });

    it('RandomFoodMapView', () => {
      const wrapper = shallow(<RandomFoodMapView />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Favourites', () => {
      const wrapper = shallow(<Favourites />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Settings', () => {
      const wrapper = shallow(<Settings />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Contact', () => {
      const wrapper = shallow(<Contact />);
      expect(wrapper).toMatchSnapshot();
    });

    it('RestaurantDetails', () => {
      const wrapper = shallow(<RestaurantDetails />);
      expect(wrapper).toMatchSnapshot();
    });

    it('FloatingActionButton', () => {
      const wrapper = shallow(<FloatingActionButton />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Snackbar', () => {
      const wrapper = shallow(<Snackbar />);
      expect(wrapper).toMatchSnapshot();
    });

    it('DisplayResult', () => {
      const wrapper = shallow(<DisplayResult />);
      expect(wrapper).toMatchSnapshot();
    });

    it('CardView', () => {
      const wrapper = shallow(<CardView />);
      expect(wrapper).toMatchSnapshot();
    });

    it('CustomMapList', () => {
      const wrapper = shallow(<CustomMapList />);
      expect(wrapper).toMatchSnapshot();
    });

    it('MyStoreCheckout', () => {
      const wrapper = shallow(<MyStoreCheckout />);
      expect(wrapper).toMatchSnapshot();
    });

    it('ReactTable', () => {
      const wrapper = shallow(<ReactTable />);
      expect(wrapper).toMatchSnapshot();
    });

    it('ImageSlider', () => {
      const wrapper = shallow(<ImageSlider />);
      expect(wrapper).toMatchSnapshot();
    });

    it('CustomMap', () => {
      const wrapper = shallow(<CustomMap />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
