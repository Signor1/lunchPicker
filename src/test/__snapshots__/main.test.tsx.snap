// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main.test render test CardView 1`] = `ShallowWrapper {}`;

exports[`main.test render test Contact 1`] = `ShallowWrapper {}`;

exports[`main.test render test CustomMap 1`] = `ShallowWrapper {}`;

exports[`main.test render test CustomMapList 1`] = `ShallowWrapper {}`;

exports[`main.test render test DisplayResult 1`] = `ShallowWrapper {}`;

exports[`main.test render test Favourites 1`] = `ShallowWrapper {}`;

exports[`main.test render test FloatingActionButton 1`] = `ShallowWrapper {}`;

exports[`main.test render test ImageSlider 1`] = `ShallowWrapper {}`;

exports[`main.test render test MyStoreCheckout 1`] = `ShallowWrapper {}`;

exports[`main.test render test NavBar 1`] = `ShallowWrapper {}`;

exports[`main.test render test RandomFood 1`] = `ShallowWrapper {}`;

exports[`main.test render test RandomFoodMapView 1`] = `ShallowWrapper {}`;

exports[`main.test render test ReactTable 1`] = `ShallowWrapper {}`;

exports[`main.test render test RestaurantDetails 1`] = `ShallowWrapper {}`;

exports[`main.test render test Settings 1`] = `ShallowWrapper {}`;

exports[`main.test render test Snackbar 1`] = `ShallowWrapper {}`;
